app/models/employer_events/renderer.rb

```diff
@@ -54,6 +54,55 @@ def renewal_and_no_future_plan_year?(carrier)
       !found_future_plan_year
     end
 
+    def find_plan_year_dates(carrier, date)
+      dates = carrier_plan_years(carrier).map do |node|
+      node.xpath("cv:#{date}", {:cv => XML_NS}).map do |date_node|
+         Date.strptime(date_node.content, "%Y%m%d") rescue nil
+        end
+      end    
+      dates.flatten.sort.last if dates.present? rescue nil
+    end
+    
+    def found_previous_plan_year_for_employer?(employer,plan_year)
+      employer.plan_years.detect{|py|py.end_date == (plan_year.start_date - 1.day)}.present? 
+    end
+
+    def qualifies_to_update_event_name?(carrier, employer_event)
+      events = [ EmployerEvents::EventNames::RENEWAL_SUCCESSFUL_EVENT, EmployerEvents::EventNames::FIRST_TIME_EMPLOYER_EVENT_NAME ]
+      return false unless employer_event.event_name.in?(events)
+      return false unless carrier.uses_issuer_centric_sponsor_cycles
+    end
+
+    def update_event_name(carrier, employer_event)
+      if qualifies_to_update_event_name?(carrier, employer_event) && found_previous_plan_year_for_carrier?(carrier)
```

Comment:
You can probably save yourself some trouble and return right off the bat using a guard clause.

It will cleanly eliminate the dangling one-statement else blah end block, as well as make it clear what the quick and common path is.